client/src/components/App.js

```diff
@@ -5,22 +5,71 @@ import NavHeader from "./NavHeader";
 import Creator from "./Creator";
 import Example from "./Example";
 import Fan from "./Fan";
+import Web3 from "web3";
+import Web3Modal from "web3modal";
+import Fortmatic from "fortmatic";
+
+let provider = null;
+let web3Modal = null;
+let web3 = null;
+let accounts = null;
+let mainAddress = "";
+
+const providerOptions = {
+  fortmatic: {
+    package: Fortmatic, // required
+    options: {
+      key: process.env.FORTMATIC_KEY, // required
+    },
+  },
+};
 
 class App extends Component {
   constructor(props) {
     super(props);
+    this.state = {
+      connectedWallet: "",
+    };
     this.openWalletConnectModal = this.openWalletConnectModal.bind(this);
   }
 
-  openWalletConnectModal() {
-    // TODO: Open web3modal and set connectedWallet when rendering Fan.
-    console.log("Opened wallet connect");
+  async openWalletConnectModal() {
+    if (!provider) {
+      web3Modal = new Web3Modal({
+        cacheProvider: true, // optional
+        providerOptions, // required
+      });
+      provider = await web3Modal.connect();
+      web3 = new Web3(provider);
+    }
+    if (!accounts) {
+      accounts = await web3.eth.getAccounts();
+      mainAddress = accounts[0].toLowerCase();
+      this.setState({
+        connectedWallet: mainAddress,
+      });
+    }
+  }
+  async disconnectWallet() {
+    provider = null;
+    accounts = null;
+    web3 = null;
+    await web3Modal.clearCachedProvider();
+    web3Modal = null;
+    mainAddress = "";
+    this.setState({
+      connectedWallet: "",
+    });
```

Comment:
When everything's in `state`, this becomes:
```
async disconnectWallet() {
// Additional part from web3modal example
  const { web3 } = this.state;
  if (web3 && web3.currentProvider && web3.currentProvider.close) {
    await web3.currentProvider.close();
  }
// what our code above becomes
  await this.web3Modal.clearCachedProvider();
  this.setState({ ...INITIAL_STATE });
}
```
From: https://github.com/Web3Modal/web3modal/blob/master/example/src/App.tsx#L536-L543